src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -64,49 +65,49 @@
    *
    * @param bigQuery the bigquery client for the project
    * @param storage the storage client for the project
-   * @param datasetName the name of the dataset
+   * @param datasetId the name of the dataset
    * @param bucketName the name of the bucket
    * @param location the location of the resources, this is only applied if both the bucket and dataset do not exist
    * @param cmekKey the name of the cmek key
    * @throws IOException if there was an error creating or fetching any GCP resource
    */
   public static void createResources(BigQuery bigQuery, Storage storage,
-                                     String datasetName, String bucketName, @Nullable String location,
+                                     DatasetId datasetId, String bucketName, @Nullable String location,
                                      @Nullable String cmekKey) throws IOException {
-    Dataset dataset = bigQuery.getDataset(datasetName);
+    Dataset dataset = bigQuery.getDataset(datasetId);
     Bucket bucket = storage.get(bucketName);
 
     if (dataset == null && bucket == null) {
       createBucket(storage, bucketName, location, cmekKey,
                    () -> String.format("Unable to create Cloud Storage bucket '%s'", bucketName));
-      createDataset(bigQuery, datasetName, location,
-                    () -> String.format("Unable to create BigQuery dataset '%s'", datasetName));
+      createDataset(bigQuery, datasetId, location,
+                    () -> String.format("Unable to create BigQuery dataset '%s'", datasetId));
```

Comment:
I think if would be useful to print both the dataset name and project. I also don't like that we're simply using that the `toString()` method for DatasetID returns.

Reply:
done:)